lib/semantics/assignment.cc

```diff
@@ -113,21 +113,19 @@ template<typename T>
 void CheckPointerAssignment(parser::ContextualMessages &messages,
     const IntrinsicProcTable &intrinsics, const Symbol &lhs,
     const Designator<T> &d) {
-  if (const auto *symbol{d.GetBaseObject().symbol()}) {
-    const Symbol &ultimate{symbol->GetUltimate()};
+  const Symbol *last{d.GetLastSymbol()}, *base{d.GetBaseObject().symbol()};
```

Comment:
It's just a matter of opinion, but I prefer one declaration per line, especially when they are initialized.

Reply:
As you wish.